core/shared/src/main/scala/zio/prelude/ForEach.scala

```diff
@@ -416,6 +426,8 @@ trait ForEachSyntax {
       F.forEach_(self)(f)
     def isEmpty(implicit F: ForEach[F]): Boolean                                                                =
       F.isEmpty(self)
+    def intercalate[A](fa: F[A], middle: A)(implicit I: Identity[A]): A                                         =
```

Comment:
We don't want to take an `fa` here but instead should call this on `self` from above.

Reply:
Thank you for reviewing this PR so quickly. And I'm sorry I rushed it too fast. I should've caught it myself. I'll fix it up.